rust/cas_client/src/caching_client.rs

```diff
@@ -77,35 +72,34 @@ impl<T: Client + Debug + Sync + Send> Client for CachingClient<T> {
         hash: &MerkleHash,
         data: Vec<u8>,
         chunk_boundaries: Vec<u64>,
-    ) -> Result<(), CasClientError> {
+    ) -> Result<()> {
         // puts write through
         debug!(
             "CachingClient put to {}/{} of length {} bytes",
             prefix,
             hash,
             data.len()
         );
-        self.client.put(prefix, hash, data, chunk_boundaries).await
+        Ok(self
+            .client
+            .put(prefix, hash, data, chunk_boundaries)
+            .await?)
     }
 
-    async fn flush(&self) -> Result<(), CasClientError> {
+    async fn flush(&self) -> Result<()> {
         // forward flush to the underlying client
-        self.client.flush().await
+        Ok(self.client.flush().await?)
     }
```

Comment:
same as above this is weird to me, I'd rather a map_err than unwrap and rewrap an ok

Reply:
See comment above.